### ISSUE TITLE
Enhancing contribution processes

### ISSUE THREAD

**Issue by oriyalperin** [JThh/fair-alloc-app-ra#7]
Hey,

@erelsgl and I have discussed the possibility of making further contributions to this repository. 
Several ideas arose:

1. There is a need for a clearer contribution guide, primarily for the branches section.
2. Is it possible to enhance the readability and reusability of the app's pages? For instance, by separating the HTML and creating an interface or library for common elements, such as the preferences table and basic inputs.

We hope to continue contributing to your fantastic app!
Thanks once again.

**Comment by JThh** [comment]
I agree with you on your proposed enhancements. They all sound reasonable to me. It would be great to see your further contributions to this project. I will get to help review the PR and also contribute codes if necessary. 

Again appreciate your interest and efforts!

**Comment by oriyalperin** [comment]
Ok, so we need a clarification regarding the branches:
1. What's the role of the `main` branch? When should we use it?
2. To which branch should changes be added and submitted? (for the PR)
3. Is the `new_main` branch used for the deployed app?

**Comment by JThh** [comment]
Thanks for your question!

1. The `main` is actually not linked to the app deployment (`new_main` does). We left it there since we wanted to deploy the app via some other (fees-charging) cloud providers, while for now we still stick to the free Streamlit platform utilities. So our PRs will only eventually be pushed to the `new_main` branch to take effect. You may ignore the `main` branch for now. 
2. For PR, it would be the best to make to the `dev` branch, which is linked to a [test app](https://fair-alloc-app-ra-7h8nhfqiswwfe2qrkksckj.streamlit.app/) for verification.
3. Yes.

**Comment by oriyalperin** [comment]
Thank you for your answers!
May I ask why you haven't set the `default` branch to be `dev`?

**Comment by JThh** [comment]
Hi, that's awesome advice and I will get to set it now.

**Comment by oriyalperin** [comment]
Great, thanks a lot! :)
